### PR TITLE
Add pedantic linter to amplify_core_plugin_interface

### PR DESCRIPTION
**Pull request by rubensdemelo**
*Description of changes:*

Add pedantic linter to amplify_core. 

https://pub.dev/packages/pedantic

### CODE REVIEW

File: packages/amplify_core_plugin_interface/lib/amplify_auth_category.dart [hunk 1]
```diff
@@ -14,64 +14,112 @@ class AuthCategory {
     // TODO: Discuss and support multiple plugins
     if (plugins.length == 0) {
       plugins.add(plugin);
-      
     } else {
-      throw("Auth plugin was not added");
+      throw ('Auth plugin was not added');
```
Comment (dariotrombello) [inline packages/amplify_core_plugin_interface/lib/amplify_auth_category.dart:18]: Reformat the file. I don't think this whitespace should be there.

**Review by jamesonwilliams — APPROVED** [review]

**Comment by Jordan-Nelson** [comment]
Hi @rubensdemelo - I posted this on one of your other PRs, but will post again here.

I think is is no longer needed as it was added to the root analysis_options.yaml
https://github.com/aws-amplify/amplify-flutter/blob/4337cd19fa62b57980e44966885ccd4fff9aa765/analysis_options.yaml#L1

Would you agree?

**Comment by rubensdemelo** [comment]
> Hi @rubensdemelo - I posted this on one of your other PRs, but will post again here.
> 
> I think is is no longer needed as it was added to the root analysis_options.yaml
> https://github.com/aws-amplify/amplify-flutter/blob/4337cd19fa62b57980e44966885ccd4fff9aa765/analysis_options.yaml#L1
> 
> Would you agree?

Yep. Thank you

**Comment by rubensdemelo** [comment]
pedantic is now deprecated.

https://pub.dev/packages/pedantic

**Pull request closed without merging**